File: python/2.7/decorator/Whip.py
from CondimentDecorator import *

class Whip(CondimentDecorator):
    def __init__(self, beverage):
        self.beverage = beverage

    def getDescription(self):
        return self.beverage.getDescription() + ", Whip"

    def cost(self):
        return .10 + self.beverage.cost()

File: js/node/decorator/com/starbuzz/condimentDecorator/Whip.js
require('./CondimentDecorator');

Whip = function(beverage) {
    CondimentDecorator.apply(this);
    this.beverage = beverage;
}
Whip.prototype = new CondimentDecorator();
Whip.prototype.getDescription = function() {
    return this.beverage.getDescription() + ", Whip"
}
Whip.prototype.cost = function() {
    return .10 + this.beverage.cost();
}